packages/flutter/lib/src/material/page.dart

```diff
@@ -233,6 +248,10 @@ class MaterialPageRoute<T> extends PageRoute<T> {
         );
       case TargetPlatform.iOS:
         return new _CupertinoPageTransition(
+          // Use a linear curve when controlled by a user gesture. This ensures
+          // the animation tracks the user's finger 1:1.
+          // See https://github.com/flutter/flutter/issues/5664
+          curve: userGesture ? null : Curves.fastOutSlowIn,
```

Comment:
I didn't implement fling. I'll have to add support for that. (It just treats fling same as stop-and-release.)

userGesture is true if the transition was started with a user gesture. It remains true throughout the transition, even after release.